subsys/caf/modules/sensor_sampler.c

```diff
@@ -134,35 +153,34 @@ static bool is_active(const struct sensor_config *sc, struct sensor_data *sd,
 	return is_active;
 }
 
-static bool can_sensor_sleep(const struct sensor_config *sc,
-			     struct sensor_data *sd,
-			     const float *curr)
+static void process_sensor_activity(const struct sensor_config *sc,
+				    struct sensor_data *sd,
+				    const float *curr)
 {
 	size_t data_cnt = get_sensor_data_cnt(sc);
 	bool sleep = true;
 
 	for (size_t i = 0; i < data_cnt; i++) {
-		if (is_active(sc, sd, curr[i], sd->prev[i])) {
+		if (process_sensor_trigger_values(sc, sd, curr[i], sd->prev[i])) {
 			sleep = false;
 			break;
 		}
 	}
 
 	if (sleep) {
-		unsigned int timeout = sc->trigger->activation.timeout_ms;
-		unsigned int period = sc->sampling_period_ms;
-
-		sd->sleep_cnt++;
-		if (sd->sleep_cnt >= timeout/period) {
-			sd->sleep_cnt = 0;
-			return true;
+		if (sd->sleep_cntd > 0) {
+			--(sd->sleep_cntd);
```

Comment:
`sd->sleep_cntd--`?

Reply:
I used to prefer predecrement over postdecrement function whenever possible as if this code would ever become c++ then it is more optimal. Just a habit. Are you sure we should use post- function here?

Reply:
prefer to post- as it seems easier to read plus it's used more frequently. but please leave as is if there are no other issues